src/Filter/Filter.php

```diff
@@ -141,4 +151,71 @@ protected function getNewParameterName(BaseProxyQueryInterface $query)
         // by underscores.
         return str_replace('.', '_', $this->getName()).'_'.$query->getUniqueParameterId();
     }
+
+    /**
+     * Adds the parameter to the corresponding `Orx` expression used in the `where` clause.
+     * If it doesn't exist, a new one is created.
+     * This method groups the filter "OR" conditions based on the "admin_code" option. If this
+     * option is not set, it uses a marker (":sonata_admin_datagrid_filter_query_marker") in
+     * the resulting DQL in order to identify the corresponding "WHERE (...)" condition
+     * group each time it is required.
+     * It allows to get queries like "WHERE previous_condition = previous_value AND (filter_1 = value OR filter_2 = value OR ...)",
+     * where the logical "OR" operators added by the filters are grouped inside a condition,
+     * instead of having unfolded "WHERE ..." clauses like "WHERE previous_condition = previous_value OR filter_1 = value OR filter_2 = value OR ...",
+     * which will produce undesired results.
+     *
+     * TODO: Remove the logic related to the ":sonata_admin_datagrid_filter_query_marker" marker when
+     * the constraint for "sonata-project/admin-bundle" guarantees that the "admin_code" option is set.
+     *
+     * @param mixed $parameter
+     */
+    private function addOrParameter(BaseProxyQueryInterface $query, $parameter): void
+    {
+        $adminCode = $this->getOption('admin_code');
```

Comment:
Even if we usually will use the admin code which a filter belongs to as value for this option, I think we could use a more agnostic name, since this option has no actual relation with an admin code; it's just a grouping identifier used to know which filters must share the same condition.
WDYT about "group" or something similar?